### ISSUE TITLE
`nvim_set_hl` raises an error with argument `fg = "fg"`

### ISSUE THREAD

**Issue by vigoux** [neovim/neovim#20008]
### Neovim version (nvim -v)

v0.8.0-dev-985-gf9c183c0c

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Not relevant

### Terminal name/version

Not relevant

### $TERM environment variable

Not relevant

### Installation

Built from source

### How to reproduce the issue

With `init.lua` being:
```lua
vim.api.nvim_set_hl(0, "Foobar", { fg = 'fg' })
```

Do:
```
nvim --clean -u init.lua
```

### Expected behavior

As stated in the documentation of `nvim_set_hl` the `"fg"` and `"bg"` values should be usable.

### Actual behavior

They are not.

```
Error detected while processing .../init.lua:
E5113: Error while calling lua chunk: .../init.lua:1: 'fg' is not a valid color
stack traceback:
        [C]: in function 'nvim_set_hl'
        .../init.lua:1: in main chunk

```

**Comment by clason** [comment]
>     Note:
        The fg and bg keys also accept the string values `"fg"` or `"bg"`
        which act as aliases to the corresponding foreground and background
        values of the Normal group. **If the Normal group has not been defined,
        using these values results in an error.**

...which I believe is not the case with `--clean`.

**Comment by vigoux** [comment]
Huh, then that's a feature and not a bug ? too bad..

**Comment by vigoux** [comment]
Weirdly though, doing `echo hlexists("Normal")` in `nvim --clean` returns 1.

**Comment by clason** [comment]
It exists but is cleared.

("You cannot just pretend that "Normal" is just like any other group"...)

**Comment by justinmk** [comment]
Is anything actionable here? Does https://github.com/neovim/neovim/pull/37546 help?

**Comment by clason** [comment]
No, but the new default colorscheme defines `Normal` so the `'fg'` and `'bg'` are now set even with `--clean` so this no longer errors.
